src/build.cc

```diff
@@ -618,12 +629,17 @@ void RealCommandRunner::Abort() {
   subprocs_.Clear();
 }
 
-size_t RealCommandRunner::CanRunMore() const {
+size_t RealCommandRunner::CanRunMore(bool jobserver_enabled) const {
   size_t subproc_number =
       subprocs_.running_.size() + subprocs_.finished_.size();
 
   int64_t capacity = config_.parallelism - subproc_number;
 
+  // Return "infinite" capacity if a jobserver is used to limit the number
+  // of parallel subprocesses instead.
+  if (jobserver_enabled)
+    return SIZE_MAX;
+
```

Comment:
Can  you explain why this is needed here inside the function, instead of the caller doing the check itself? This would avoid adding one extra parameter to the method in the virtual base class, and other derived implementations.

Reply:
I considered making the call to `CanRunMore()` conditional on `Jobserver::Enabled()` but decided to avoid the extra indentation / complexity in `Builder::Build()` method. Especially because I think the `Build()` method is already quite hard to follow.